src/transformers/custom-elements-manifest/utils.ts

```diff
@@ -0,0 +1,180 @@
+import { getTypeHintFromType } from "../../util/get-type-hint-from-type";
+
+/**
+ * Returns a Reference to a node
+ * @param node
+ * @param context
+ */
+export function getReferenceForNode(node: Node, context: TransformerContext): schema.Reference {
+	const sourceFile = node.getSourceFile();
+	const name = getNodeName(node, context) as string;
+
+	// Test if the source file is from a typescript lib
+	// TODO: Find a better way of checking this
+	const isLib = sourceFile.isDeclarationFile && sourceFile.fileName.match(/typescript\/lib.*\.d\.ts$/) != null;
+	if (isLib) {
+		// Only return the name of the declaration if it's from lib
+		return {
+			name
+		};
+	}
+
+	// Test if the source file is located in a package
+	const packageName = getPackageName(sourceFile);
+	if (packageName != null) {
+		return {
+			name,
+			package: packageName
+		};
+	}
+
+	// Get the module path name
+	const module = getRelativePath(sourceFile.fileName, context);
+	return {
+		name,
+		module
+	};
+}
+
+export function getInheritedFromReference(
+	onDeclaration: ComponentDeclaration,
+	feature: ComponentFeatureBase,
+	context: TransformerContext
+): schema.Reference | undefined {
+	if (feature.declaration != null && feature.declaration !== onDeclaration) {
+		return getReferenceForNode(feature.declaration.node, context);
+	}
+
+	return undefined;
+}
+
+/**
+ * Returns a relative path based on "cwd" in the config
+ * @param fullPath
+ * @param context
+ */
+export function getRelativePath(fullPath: string, context: TransformerContext): string {
+	return context.config.cwd != null ? `./${relative(context.config.cwd, fullPath)}` : basename(fullPath);
+}
+
+/**
+ * Returns the name of the package (if any)
+ * @param sourceFile
+ */
+export function getPackageName(sourceFile: SourceFile): string | undefined {
+	// TODO: Make it possible to access the ModuleResolutionHost
+	//  in order to resolve the package using "resolveModuleNames"
+	//  The following approach is very, very naive and is only temporary.
+	const match = sourceFile.fileName.match(/node_modules\/(.*?)\//);
+
+	if (match != null) {
+		return match[1];
+	}
+
+	return undefined;
+}
+
+/**
+ * Returns description and typeHint based on jsdoc for a specific parameter name
+ * @param name
+ * @param jsDoc
+ */
+export function getParameterFromJsDoc(name: string, jsDoc: JsDoc | undefined): { description?: string; typeHint?: string } {
+	if (jsDoc?.tags == undefined) {
+		return {};
+	}
+
+	for (const tag of jsDoc.tags) {
+		const parsed = tag.parsed();
+
+		if (parsed.tag === "param" && parsed.name === name) {
+			return { description: parsed.description, typeHint: parsed.type };
+		}
+	}
+
+	return {};
+}
+
+/**
+ * Get return description and return typeHint from jsdoc
+ * @param jsDoc
+ */
+export function getReturnFromJsDoc(jsDoc: JsDoc | undefined): { description?: string; typeHint?: string } {
+	const tag = jsDoc?.tags?.find(tag => ["returns", "return"].includes(tag.tag));
```

Comment:
```suggestion
	const tag = jsDoc?.tags?.find(tag => tag === "returns" || tag === "return");
```

Comparison is going to be a lot faster and it's possibly easier to read IMO.